app/models/rubric_criterion.rb

```diff
@@ -167,14 +171,16 @@ def self.load_from_yml(criterion_yml)
     # Visibility options
     criterion.ta_visible = criterion_yml[1]['ta_visible'] unless criterion_yml[1]['ta_visible'].nil?
     criterion.peer_visible = criterion_yml[1]['peer_visible'] unless criterion_yml[1]['peer_visible'].nil?
-
+    levels = []
     criterion_yml[1]['levels'].each do |name, level_yml|
-      criterion.levels.build(rubric_criterion: criterion,
-                             name: name,
-                             description: level_yml['description'],
-                             mark: level_yml['mark'])
+      level = Level.new
```

Comment:
You should be able to keep the exact same code as before but just append it directly to `levels` using `<<`, i.e. `levels << criterion.levels.build(...)`.